hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -150,16 +154,37 @@ public ClientConnectionManagerImpl(HazelcastClientInstanceImpl client, AddressTr
     }
 
     protected void initIOThreads(HazelcastClientInstanceImpl client) {
-        boolean directBuffer = client.getProperties().getBoolean(SOCKET_CLIENT_BUFFER_DIRECT);
+        HazelcastProperties properties = client.getProperties();
+        boolean directBuffer = properties.getBoolean(SOCKET_CLIENT_BUFFER_DIRECT);
+
+        SSLConfig sslConfig = client.getClientConfig().getNetworkConfig().getSSLConfig();
+        boolean sslEnabled = sslConfig != null && sslConfig.isEnabled();
+
+        int configuredInputThreads = properties.getInteger(ClientProperty.IO_INPUT_THREAD_COUNT);
+        int configuredOutputThreads = properties.getInteger(ClientProperty.IO_OUTPUT_THREAD_COUNT);
+
+        int inputThreads;
+        if (configuredInputThreads == -1) {
+            inputThreads = sslEnabled ? DEFAULT_SSL_THREAD_COUNT : 1;
+        } else {
+            inputThreads = configuredInputThreads;
+        }
+
+        int outputThreads;
+        if (configuredOutputThreads == -1) {
+            outputThreads = sslEnabled ? DEFAULT_SSL_THREAD_COUNT : 1;
+        } else {
+            outputThreads = configuredOutputThreads;
+        }
 
         ioThreadingModel = new NonBlockingIOThreadingModel(
                 client.getLoggingService(),
                 client.getMetricsRegistry(),
                 new HazelcastThreadGroup(client.getName(), logger, client.getClientConfig().getClassLoader()),
                 outOfMemoryHandler,
-                1,
-                1,
-                0,
+                inputThreads,
+                outputThreads,
+                properties.getInteger(ClientProperty.IO_BALANCER_INTERVAL_SECONDS),
```

Comment:
If there is only a single thread in the io system, the balancer disabled itself (nothing to do). The 20 seconds are the same as on the server side.

No validation check,but we don't do on the server side:

```
  @Override
    public int getBalancerIntervalSeconds() {
        return node.getProperties().getSeconds(GroupProperty.IO_BALANCER_INTERVAL_SECONDS);
    }
```